### PR TITLE
Adding DogStatsD docs

### PR DESCRIPTION
**Pull request by scott-shields-github**
Adding documentation for DogStatsD codec.
PR - https://github.com/tremor-rs/tremor-runtime/pull/1957

To be merged once the previous PR is merged.

Signed-off-by: Scott Shields <sshields@wayfair.com>

### CODE REVIEW

File: versioned_docs/version-0.12/reference/codecs/dogstatsd.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# dogstatsd
+
+The `dogstatsd` codec supports the [Datadog DogStatsD v1.2 protocol](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell?tab=metrics).
+
+The format is similar the `statsd` format, but also includes events and service checks.
```
Comment (mfelsche) [inline versioned_docs/version-0.12/reference/codecs/dogstatsd.md:5]:
```suggestion
The format is similar to the `statsd` format, but also includes events and service checks.
```

File: versioned_docs/version-0.12/reference/codecs/dogstatsd.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# dogstatsd
+
+The `dogstatsd` codec supports the [Datadog DogStatsD v1.2 protocol](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell?tab=metrics).
```
Comment (mfelsche) [inline versioned_docs/version-0.12/reference/codecs/dogstatsd.md:3]:
```suggestion
The `dogstatsd` codec supports the [Datadog DogStatsD v1.2 protocol](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell).
```

File: versioned_docs/version-0.12/reference/codecs/dogstatsd.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+The format is similar the `statsd` format, but also includes events and service checks.
+
+The codec translates a single `dogstatsd` measurement line into a structured event.
```
Comment (mfelsche) [inline versioned_docs/version-0.12/reference/codecs/dogstatsd.md:7]:
```suggestion
The codec translates a single `dogstatsd` measurement line into a structured event and vice versa.
```

File: versioned_docs/version-0.12/reference/codecs/dogstatsd.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+The equivalent representation as a tremor value:
+
+```json
+{
+  "metric": {
+    "type": "c",
+    "metric": "datadog.metric",
+    "value": 7,
```
Comment (mfelsche) [inline versioned_docs/version-0.12/reference/codecs/dogstatsd.md:26]:
```suggestion
    "values": [7],
```

**Review by mfelsche — DISMISSED** [review]
Some small nitpicks, but otherwise LGTM! Thank you!

**Review by darach — APPROVED** [review]
LGTM 👍

**Review by mfelsche — DISMISSED** [review]

**Comment by scott-shields-github** [comment]
Moved these out of the v0.12 docs since it will available in 0.13

**Review by mfelsche — APPROVED** [review]

**Pull request merged**